kronofoto/static/assets/js/lib.js

```diff
@@ -18,633 +18,808 @@ import { Tooltip } from 'foundation-sites/js/foundation.tooltip';
 
-  htmx.onLoad((e) => {
-    if(typeof window.kfcontext.onceInitHTMXListeners === 'undefined') {
-      window.kfcontext.onceInitHTMXListeners = true
-      initTimeline(context)
     }
-    initPopups(e)
-    if (window.st && e.querySelectorAll('.sharethis-inline-share-buttons').length) {
-        st.initialize()
+    else {
+      self.stylesReady = true
+      self.tryInitTimline()
+      self.initHTMXListeners(self.context, self.root)
+      self.initFoundation(self.context)
+      self.initGalleryNav()
+      self.initClipboardJS(self.context)
+      self.initEventHandlers(self.context)
+      self.installButtons(self.context)(self.root)
+      self.initPopups()
+      self.initShareThis()
+      self.initDraggableThumbnails()
+      self.initNavSearch()
+      self.initAutocomplete()
     }
+  }
 
-    // Init gallery thumbnails
-    if ($('#fi-preload-zone li').length) {
-
-      initDraggableThumbnails()
-
-      let html = $('#fi-preload-zone').html()
-      // let firstId = $(html).find('li:first-child span').attr('hx-get')
-      $('#fi-thumbnail-carousel-images').html(html)
-      // let $after = $('#fi-thumbnail-carousel-images li span[hx-get="'+firstId+'"]').nextAll()
-      // $('#fi-thumbnail-carousel-images').append($after)
-      $('#fi-thumbnail-carousel-images').addClass('dragging')
-      $('#fi-thumbnail-carousel-images').css('left', '0px')
-      setTimeout(() => {
-        $('#fi-thumbnail-carousel-images').removeClass('dragging')
-        if (!$('#fi-thumbnail-carousel-images [data-origin]').length) {
-          $('#fi-thumbnail-carousel-images [data-active]').attr('data-origin', '')
-        }
-      }, 100)
+  jquery(sel, _context) {
+    var self = this
+    _context = _context || self.context
+    return $(sel, _context)
+  }
 
-      htmx.process($('#fi-thumbnail-carousel-images').get(0))
-      $('#fi-preload-zone').empty()
+  // initJQuery(context) {
+  //
+  //   let self = this
+  //   var initjQuery = $.fn.init
+  //   self.jQuery.fn.init = function (s, c, r) {
+  //     c = c || context
+  //     return new initjQuery(s, c, r)
+  //   };
+  //   $.fn.init = self.jQuery.fn.init
+  // }
+
+  tryInitTimline() {
+    let self = this
+    if(self.stylesReady && self.htmxLoaded && self.jquery('.photos-timeline').length && !self.jquery('.photos-timeline').hasClass('initialized')) {
+      self.jquery('.photos-timeline').addClass('initialized')
+      self.initTimeline()
     }
+  }
 
-    if ($('#fi-image-preload img').length && !$('#fi-image-preload img').data('loaded')) {
-      $('#fi-image-preload img').data('loaded', true)
-      let url = $('#fi-image-preload img').attr('src')
-      const image = new Image();
-      image.src = url;
-      image.onload = () => {
-        let html = $('#fi-image-preload').html()
-        $('#fi-image').html(html)
-        $('#fi-image-preload').empty()
+  initShareThis() {
+    let self = this
+    setInterval(() => {
```

Comment:
This came from a conflict resolution. I didn't have this in my latest code, but added it because it was in the code I was pulling in. I agree, I don't like it either. I was just trying to get everything merged. I just wasn't able to test it locally post-merge because the server wouldn't start up. I'd like to get rid of that assuming it works without it.